packages/integration-tests/src/EntityManager.populate.test.ts

```diff
@@ -80,6 +80,81 @@ describe("EntityManager.populate", () => {
     expect(book.tags.get.length).toEqual(0);
   });
 
+  it("can populate via promise on direct load", async () => {
+    // Given a book with an author and no tags
+    await insertAuthor({ first_name: "a1" });
+    await insertBook({ title: "b1", author_id: 1 });
+
+    // When we load and then populate author and tags
+    const em = newEntityManager();
+    const book = await em.load(Book, "1").populate(["author", "tags"]);
+
+    // we can access author and tags but no tags will be present
+    expect(book.author.get.idOrFail).toEqual("a:1");
+    expect(book.tags.get.length).toEqual(0);
+  });
+
+  it("can propagate undefined through a populate promise", async () => {
+    // Given an author with no publisher
+    await insertAuthor({ first_name: "a1" });
+
+    // When we load the author then try to load and populate through publisher
+    const em = newEntityManager();
+    const author = await em.load(Author, "a:1");
+    const publisher = await author.publisher.load().populate("images");
```

Comment:
And this is what we originally wanted to be `author.publisher.load("images")` but doesn't jive with TS? 

Can you just the return type of `Reference.load` to `Promise<...> & { populate: ... }`, and then monkey-patch `populate` directly onto that single `promise` instead of the global `Promise` prototype?

I guess I wouldn't be surprised if, to do something like:

```
const promise = em.load(....);
promise.populate = ....
return promise;
```

We can't use `async` functions b/c the compiler/runtime rewriting of the function might create it's own promise, and not use the promise that we monkey-patched with a `populate` key.

Reply:
iirc adding a `Loaded` or `LoadHint` anywhere in the definition of `Reference` or `Collection` resulted in circular reference compiler errors